starboard/nplb/posix_compliance/posix_socket_helpers.h

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_NPLB_POSIX_COMPLIANCE_POSIX_SOCKET_HELPERS_H_
+#define STARBOARD_NPLB_POSIX_COMPLIANCE_POSIX_SOCKET_HELPERS_H_
+
+#include <fcntl.h>
```

Comment:
nit: like we talked about offline, I think including headers here based on what the source files need to include is antithetical to the IWYU practice

Reply:
The <fcntl.h> is used by 2 send related tests, so <fcntl.h> is removed from helpers and only gets included in those 2.

socket_create is the only test that doesn't have any dependencies on those helper functions, so it doesn't need to include the helpers.h.